### PR TITLE
Update loot tracker to sort drops by highest value

### PR DESCRIPTION
**Pull request by arthurtyukayev**
Closes https://github.com/runelite/runelite.net/issues/213

### CODE REVIEW

File: src/modules/loot.js [hunk 1]
```diff
@@ -127,45 +127,45 @@ export const getGroupedLoot = createSelector(
       const items = [...existingDrops].concat(newDrops)
       const groupedItems = []
 
-      for (let item of items) {
+      items.forEach(item => {
```
Comment (Nightfirecat) [inline src/modules/loot.js]: Why are these unrelated lines changed for what is essentially a personal preference?

Comment (deathbeam) [reply src/modules/loot.js]: I agree this forEach changes should be reverted. In some cases it is even changing the functionality

Comment (arthurtyukayev) [reply src/modules/loot.js]: Fixed.

File: src/modules/loot.js [hunk 2]
```diff
@@ -178,6 +178,23 @@ export const getGroupedLoot = createSelector(
       groupedLoot.set(key, newEntry)
     }
 
+    // eslint-disable-next-line no-unused-vars
+    for (let [name, { drops }] of groupedLoot) {
```
Comment (Nightfirecat) [inline src/modules/loot.js]: These can become `for (let { drops } of groupedLoot.values()) {` to remove the unused var instead of adding an eslint-disable directive.

File: src/modules/loot.js [hunk 3]
```diff
@@ -178,6 +178,23 @@ export const getGroupedLoot = createSelector(
...
+      const formatPrice = drop => {
+        let price = prices[drop.id] * drop.qty
+        if (isNaN(price)) {
+          price = drop.qty
+        }
+        return price
+      }
+
+      if (drops) {
```
Comment (Nightfirecat) [inline src/modules/loot.js]: If this `if` statement is needed, couldn't this loop be written as follows? (that is, with a guard clause instead of a guarding `if` statement)

```
for (...) {
  if (!drops) {
    continue
  }
  
  const formatPrice = ...

  for (let drop of drops) {
    drop.price = formatPrice(drop)
  }
}
```

Comment (arthurtyukayev) [reply src/modules/loot.js]: I was just being cautious. But I don't think it's necessary. I've removed that check. I've also moved the `formatPrice` function outside of the for-loop.

**Comment by Adam-** [comment]
@deathbeam can you look at this?

File: src/routes/account/loot-tracker.js [hunk 4]
```diff
@@ -99,19 +99,25 @@ const formatPrice = price => {
...
-      {record.name}
-      <span class="small float-right">
-        x {record.count} ({formatPrice(record.price)}gp)
-      </span>
-    </div>
-    <div class="card-body pt-0 pb-0 record-body">
-      <div class="row drop-row">{record.drops.map(buildDrop)}</div>
+const buildLootRecord = record => {
+  const buildDrops = drops => {
+    return drops.sort((a, b) => b.price - a.price).map(buildDrop)
```
Comment (deathbeam) [inline src/routes/account/loot-tracker.js]: The drops should be sorted in selector

**Comment by deathbeam** [comment]
Any update on this?

**Comment by arthurtyukayev** [comment]
@deathbeam Missed the updates on this. Should have some time to work on this by the end of the week.

**Comment by deathbeam** [comment]
Alright, thanks
